src/alire/alire-directories.adb

```diff
@@ -92,6 +133,10 @@ package body Alire.Directories is
            Kind (Possible_Root.Crate_File) = Ordinary_File
          then
             return Path;
+         elsif GNAT.OS_Lib.Is_Directory ("alire") and then
+           Directories.Find_Single_File ("alire", "toml") /= ""
```

Comment:
Will this match with `config.toml`?

Reply:
For an old workspace no, because it won't exist, and in a new one it won't reach that branch because it will have detected `alire.toml` in the previous condition.

But since I haven't tested this exhaustively I'm going to add a test to exclude that possibility just in case.